src/billing_infos/swico/mod.rs

```diff
@@ -0,0 +1,156 @@
+//! FRENCH: https://www.swiss-qr-invoice.org/downloads/qr-bill-s1-syntax-fr.pdf
+//! GERMAN: https://www.swiss-qr-invoice.org/downloads/qr-bill-s1-syntax-de.pdf
+//!
+
+use crate::billing_infos::{utils::TotalLenght, BillingInfos, Emitter, RawData, RawDataKind};
+use crate::NaiveDate;
+use std::{collections::BTreeMap, fmt::Display, sync::Arc};
+
+mod parser;
+use parser::s1_parser;
+mod builder;
+use builder::S1Builder;
+mod erro;
+pub use erro::SwicoError;
+mod syntax;
+use syntax::Version;
+
+const DATE_FMT: &str = "%y%m%d";
+
+type StructuredSet = BTreeMap<SwicoComponent, Arc<str>>;
```

Comment:
```suggestion
type StructuredSet = BTreeMap<SwicoComponent, &'str>;
```

I think it should be possible to not use an `Arc`. If anything I would use a `Cow` :)